use issn rather than journal name for url; improve page load time

rnathanday opened the pull request:
This update to the journal-landing module includes:
- for /pages/journalLookup page, move table to static json file; if issn available, add link from journal title to /journal/nnnn-nnnn page
- update journal lookup table to dfagandryad:LookupPaleoBiology
- move journal-landing db queries into schema from java src
- allow solr server to be injected in DryadJournal class for ease of testing
- execute solr queries in parallel using cinclude transformer
- add AMJBOT.png for journal landing page
- add journal-landing module to xmlui aspect chain
- update max number of boolean clauses in solr query for statistics core, for handling item queries

For deployment, there is a migration script that must be run: database_schema_17-18.sql

The journal-landing module contains a number of unit tests to exercise the Java code against test data. This has been deployed to the dev server.

These wiki pages have been updated:
    http://wiki.datadryad.org/Journal_Landing_Pages_build
    http://wiki.datadryad.org/Journal_Landing_Pages

rnathanday committed:
use issn rather than journal name for url; run solr queries as includes

- for /pages/journalLookup page, move table to static json file; if issn available, add link from journal title to /journal/nnnn-nnnn page
- update journal lookup table to dfagandryad:LookupPaleoBiology
- move journal-landing db queries into schema from java src
- allow solr server to be injected in DryadJournal class for ease of testing
- execute solr queries in parallel using cinclude transformer
- add AMJBOT.png for journal landing page
- add journal-landing module to xmlui aspect chain
- combine journal-landing java transformers into a single class to avoid additional dispatching during processing

ghost commented:
The most recent update fixes this line to avoid empty journal sponsor names:

https://github.com/datadryad/dryad-repo/pull/1147/commits/4e01262b3eb04758b8e30e00b6263200dfac532a#diff-4ece0424ab7ae568cdc81588bc86d6c8R193

rnathanday committed:
fix unit tests and test setup; cleanup of pipeline debugging

ghost commented:
@daisieh 0818d86 updates the tests primarily

daisieh commented:
Most of the pages seem to work fine, but I get an error on my VM when I try to load Evolution: http://localhost:9999/journal/1936-6426

```
<!--java.io.IOException: Error writing to server
    at sun.net.www.protocol.http.HttpURLConnection.writeRequests(HttpURLConnection.java:582)
    at sun.net.www.protocol.http.HttpURLConnection.writeRequests(HttpURLConnection.java:594)
    at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1216)
    at sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:2340)
    at java.net.HttpURLConnection.getHeaderFieldDate(HttpURLConnection.java:450)
    at java.net.URLConnection.getLastModified(URLConnection.java:532)
    at org.apache.excalibur.source.impl.URLSource.getInfos(URLSource.java:190)
    at org.apache.excalibur.source.impl.AbstractSource.checkInfos(AbstractSource.java:62)
    at org.apache.excalibur.source.impl.URLSource.getInputStream(URLSource.java:253)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:180)
    at org.apache.cocoon.components.source.SourceUtil.toSAX(SourceUtil.java:82)
    at org.apache.cocoon.transformation.CIncludeTransformer.processCIncludeElement(CIncludeTransformer.java:562)
    at org.apache.cocoon.transformation.CIncludeTransformer.startTransformingElement(CIncludeTransformer.java:264)
    at org.apache.cocoon.transformation.AbstractSAXTransformer.startElement(AbstractSAXTransformer.java:460)
    at sun.reflect.GeneratedMethodAccessor321.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy267.startElement(Unknown Source)
    at org.apache.xml.serializer.ToXMLSAXHandler.closeStartTag(ToXMLSAXHandler.java:204)
    at org.apache.xml.serializer.ToSAXHandler.flushPending(ToSAXHandler.java:277)
    at org.apache.xml.serializer.ToXMLSAXHandler.endElement(ToXMLSAXHandler.java:243)
    at org.apache.xalan.templates.ElemElement.constructNode(ElemElement.java:344)
    at org.apache.xalan.templates.ElemElement.execute(ElemElement.java:288)
    at org.apache.xalan.transformer.TransformerImpl.executeChildTemplates(TransformerImpl.java:2411)
    at org.apache.xalan.templates.ElemTemplate.execute(ElemTemplate.java:392)
    at org.apache.xalan.templates.ElemCallTemplate.execute(ElemCallTemplate.java:246)
    at org.apache.xalan.transformer.TransformerImpl.executeChildTemplates(TransformerImpl.java:2411)
    at org.apache.xalan.templates.ElemCopy.execute(ElemCopy.java:114)
    at org.apache.xalan.templates.ElemApplyTemplates.transformSelectedNodes(ElemApplyTemplates.java:393)
    at org.apache.xalan.templates.ElemApplyTemplates.execute(ElemApplyTemplates.java:176)
    at org.apache.xalan.transformer.TransformerImpl.executeChildTemplates(TransformerImpl.java:2411)
    at org.apache.xalan.templates.ElemCopy.execute(ElemCopy.java:114)
    at org.apache.xalan.templates.ElemApplyTemplates.transformSelectedNodes(ElemApplyTemplates.java:393)
    at org.apache.xalan.templates.ElemApplyTemplates.execute(ElemApplyTemplates.java:176)
    at org.apache.xalan.transformer.TransformerImpl.executeChildTemplates(TransformerImpl.java:2411)
    at org.apache.xalan.templates.ElemCopy.execute(ElemCopy.java:114)
    at org.apache.xalan.templates.ElemApplyTemplates.transformSelectedNodes(ElemApplyTemplates.java:393)
    at org.apache.xalan.templates.ElemApplyTemplates.execute(ElemApplyTemplates.java:176)
    at org.apache.xalan.transformer.TransformerImpl.executeChildTemplates(TransformerImpl.java:2411)
    at org.apache.xalan.transformer.TransformerImpl.applyTemplateToNode(TransformerImpl.java:2281)
    at org.apache.xalan.transformer.TransformerImpl.transformNode(TransformerImpl.java:1367)
    at org.apache.xalan.transformer.TransformerImpl.run(TransformerImpl.java:3458)
    at org.apache.xalan.transformer.TransformerHandlerImpl.endDocument(TransformerHandlerImpl.java:406)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.apache.cocoon.transformation.TraxTransformer.endDocument(TraxTransformer.java:563)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy267.endDocument(Unknown Source)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy270.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy270.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy270.endDocument(Unknown Source)
    at org.apache.cocoon.components.sax.XMLTeePipe.endDocument(XMLTeePipe.java:71)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.components.sax.XMLTeePipe.endDocument(XMLTeePipe.java:71)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.components.sax.XMLTeePipe.endDocument(XMLTeePipe.java:71)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.components.sax.XMLTeePipe.endDocument(XMLTeePipe.java:71)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.components.sax.XMLTeePipe.endDocument(XMLTeePipe.java:71)
    at org.apache.cocoon.xml.AbstractXMLPipe.endDocument(AbstractXMLPipe.java:55)
    at org.dspace.app.xmlui.wing.AbstractWingTransformer.endDocument(AbstractWingTransformer.java:112)
    at sun.reflect.GeneratedMethodAccessor325.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy269.endDocument(Unknown Source)
    at org.apache.cocoon.environment.internal.EnvironmentChanger.endDocument(EnvironmentStack.java:119)
    at org.apache.cocoon.components.sax.XMLTeePipe.endDocument(XMLTeePipe.java:71)
    at org.apache.cocoon.components.sax.AbstractXMLByteStreamInterpreter.parse(AbstractXMLByteStreamInterpreter.java:103)
    at org.apache.cocoon.components.sax.XMLByteStreamInterpreter.deserialize(XMLByteStreamInterpreter.java:44)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:324)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
    at org.apache.cocoon.components.source.impl.SitemapSource.toSAX(SitemapSource.java:362)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:111)
    at org.apache.cocoon.components.source.util.SourceUtil.parse(SourceUtil.java:294)
    at org.apache.cocoon.generation.FileGenerator.generate(FileGenerator.java:136)
    at sun.reflect.GeneratedMethodAccessor315.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy265.generate(Unknown Source)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:326)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
    at org.apache.cocoon.components.source.impl.SitemapSource.toSAX(SitemapSource.java:362)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:111)
    at org.apache.cocoon.components.source.util.SourceUtil.parse(SourceUtil.java:294)
    at org.apache.cocoon.generation.FileGenerator.generate(FileGenerator.java:136)
    at sun.reflect.GeneratedMethodAccessor315.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy265.generate(Unknown Source)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:326)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
    at org.apache.cocoon.components.source.impl.SitemapSource.toSAX(SitemapSource.java:362)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:111)
    at org.apache.cocoon.components.source.util.SourceUtil.parse(SourceUtil.java:294)
    at org.apache.cocoon.generation.FileGenerator.generate(FileGenerator.java:136)
    at sun.reflect.GeneratedMethodAccessor315.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy265.generate(Unknown Source)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:326)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
    at org.apache.cocoon.components.source.impl.SitemapSource.toSAX(SitemapSource.java:362)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:111)
    at org.apache.cocoon.components.source.util.SourceUtil.parse(SourceUtil.java:294)
    at org.apache.cocoon.generation.FileGenerator.generate(FileGenerator.java:136)
    at sun.reflect.GeneratedMethodAccessor315.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy265.generate(Unknown Source)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:326)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
    at org.apache.cocoon.components.source.impl.SitemapSource.toSAX(SitemapSource.java:362)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:111)
    at org.apache.cocoon.components.source.util.SourceUtil.parse(SourceUtil.java:294)
    at org.apache.cocoon.generation.FileGenerator.generate(FileGenerator.java:136)
    at sun.reflect.GeneratedMethodAccessor315.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy265.generate(Unknown Source)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:326)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
    at org.apache.cocoon.components.source.impl.SitemapSource.toSAX(SitemapSource.java:362)
    at org.apache.cocoon.components.source.util.SourceUtil.toSAX(SourceUtil.java:111)
    at org.apache.cocoon.components.source.util.SourceUtil.parse(SourceUtil.java:294)
    at org.apache.cocoon.generation.FileGenerator.generate(FileGenerator.java:136)
    at sun.reflect.GeneratedMethodAccessor315.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy265.generate(Unknown Source)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.processXMLPipeline(AbstractProcessingPipeline.java:544)
    at org.apache.cocoon.components.pipeline.impl.AbstractCachingProcessingPipeline.processXMLPipeline(AbstractCachingProcessingPipeline.java:273)
    at org.apache.cocoon.components.pipeline.AbstractProcessingPipeline.process(AbstractProcessingPipeline.java:750)
    at sun.reflect.GeneratedMethodAccessor316.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at org.apache.cocoon.core.container.spring.avalon.PoolableProxyHandler.invoke(PoolableProxyHandler.java:71)
    at com.sun.proxy.$Proxy264.process(Unknown Source)
```

Any ideas, @rnathanday ?

daisieh commented:
One more thing: are `journal-lookup.json` and `AMJBOT.png` being used still?

daisieh commented:
The code looks fine, I think: I'm not really familiar with a lot of the components here so I could only look at a fairly superficial level.
I occasionally get errors: sometimes the one above for Evolution, and a different one that I've gotten sometimes for Evolution and sometimes for BMC Evolutionary Biology:

```
Caused by: java.io.IOException: Server returned HTTP response code: 400 for URL: http://localhost:9999/solr/statistics/select/?q=(time:[NOW-1MONTH%20TO%20NOW]%20AND%20-isBot:true%20AND%20owningColl:1%20AND%20owningComm:1%20AND%20(owningItem:[9297%20TO%209300]%20OR%20owningItem:[9570%20TO%209575]%20OR%20owningItem:9792%20OR%20owningItem:9981%20OR%20owningItem:10219%20OR%20owningItem:10220%20OR%20owningItem:10328%20OR%20owningItem:35568%20OR%20owningItem:38571%20OR%20owningItem:[43063%20TO%2043075]%20OR%20owningItem:43585%20OR%20owningItem:47537%20OR%20owningItem:[47920%20TO%2047922]%20OR%20owningItem:[47929%20TO%2047938]%20OR%20owningItem:49927%20OR%20owningItem:50028%20OR%20owningItem:50029%20OR%20owningItem:51913%20OR%20owningItem:[53105%20TO%2053110]%20OR%20owningItem:54902%20OR%20owningItem:55317%20OR%20owningItem:55319%20OR%20owningItem:55320%20OR%20owningItem:55322%20OR%20owningItem:55324%20OR%20owningItem:56391%20OR%20owningItem:56424%20OR%20owningItem:[56426%20TO%2056432]%20OR%20owningItem:[56589%20TO%2056592]%20OR%20owningItem:57648%20OR%20owningItem:[58086%20TO%2058091]%20OR%20owningItem:59072%20OR%20owningItem:59461%20OR%20owningItem:61321%20OR%20owningItem:61322%20OR%20owningItem:[62360%20TO%2062362]%20OR%20owningItem:62509%20OR%20owningItem:62510%20OR%20owningItem:62661%20OR%20owningItem:[62732%20TO%2062736]%20OR%20owningItem:62738%20OR%20owningItem:62951%20OR%20owningItem:64433%20OR%20owningItem:[64488%20TO%2064490]%20OR%20owningItem:64870%20OR%20owningItem:64936%20OR%20owningItem:[65574%20TO%2065578]%20OR%20owningItem:65847%20OR%20owningItem:65848%20OR%20owningItem:66259%20OR%20owningItem:[66511%20TO%2066513]%20OR%20owningItem:[66515%20TO%2066517]%20OR%20owningItem:[66519%20TO%2066523]%20OR%20owningItem:70225%20OR%20owningItem:[70227%20TO%2070229]%20OR%20owningItem:70231%20OR%20owningItem:71389%20OR%20owningItem:71390%20OR%20owningItem:71925%20OR%20owningItem:71926%20OR%20owningItem:72426%20OR%20owningItem:[72428%20TO%2072430]%20OR%20owningItem:[72477%20TO%2072479]%20OR%20owningItem:72586%20OR%20owningItem:[72963%20TO%2072965]%20OR%20owningItem:73070%20OR%20owningItem:73123%20OR%20owningItem:73909%20OR%20owningItem:77587%20OR%20owningItem:78386%20OR%20owningItem:78831%20OR%20owningItem:[78994%20TO%2079004]%20OR%20owningItem:[79458%20TO%2079460]%20OR%20owningItem:79855%20OR%20owningItem:79881%20OR%20owningItem:80114%20OR%20owningItem:[80116%20TO%2080126]%20OR%20owningItem:80181%20OR%20owningItem:[80478%20TO%2080486]%20OR%20owningItem:80488%20OR%20owningItem:80489%20OR%20owningItem:80491%20OR%20owningItem:[80638%20TO%2080640]%20OR%20owningItem:80701%20OR%20owningItem:80817%20OR%20owningItem:80949%20OR%20owningItem:81036%20OR%20owningItem:81041%20OR%20owningItem:81078%20OR%20owningItem:81081%20OR%20owningItem:81082%20OR%20owningItem:81291%20OR%20owningItem:[81340%20TO%2081343]%20OR%20owningItem:[81406%20TO%2081411]%20OR%20owningItem:81937%20OR%20owningItem:81966%20OR%20owningItem:81967%20OR%20owningItem:81994%20OR%20owningItem:[82061%20TO%2082064]%20OR%20owningItem:82142%20OR%20owningItem:82234%20OR%20owningItem:82235%20OR%20owningItem:82399%20OR%20owningItem:82400%20OR%20owningItem:82586%20OR%20owningItem:82587%20OR%20owningItem:82659%20OR%20owningItem:82810%20OR%20owningItem:83026%20OR%20owningItem:83064%20OR%20owningItem:[83119%20TO%2083122]%20OR%20owningItem:83178%20OR%20owningItem:83238%20OR%20owningItem:83310%20OR%20owningItem:83331%20OR%20owningItem:83414%20OR%20owningItem:83921%20OR%20owningItem:84133%20OR%20owningItem:84134%20OR%20owningItem:84165%20OR%20owningItem:84166%20OR%20owningItem:84173%20OR%20owningItem:84174%20OR%20owningItem:84289%20OR%20owningItem:[84750%20TO%2084755]%20OR%20owningItem:84818%20OR%20owningItem:84819%20OR%20owningItem:84832%20OR%20owningItem:[85013%20TO%2085019]%20OR%20owningItem:85160%20OR%20owningItem:85161%20OR%20owningItem:85226%20OR%20owningItem:85872%20OR%20owningItem:85897%20OR%20owningItem:[85963%20TO%2085968]%20OR%20owningItem:87466%20OR%20owningItem:[87937%20TO%2087939]%20OR%20owningItem:87941%20OR%20owningItem:[87943%20TO%2087946]%20OR%20owningItem:88056%20OR%20owningItem:88310%20OR%20owningItem:88312%20OR%20owningItem:88406%20OR%20owningItem:[88467%20TO%2088469]%20OR%20owningItem:89730%20OR%20owningItem:91248%20OR%20owningItem:91249%20OR%20owningItem:91252%20OR%20owningItem:92537%20OR%20owningItem:92538%20OR%20owningItem:94054%20OR%20owningItem:94057%20OR%20owningItem:94058%20OR%20owningItem:94172%20OR%20owningItem:[94360%20TO%2094363]%20OR%20owningItem:94883%20OR%20owningItem:94900%20OR%20owningItem:94901%20OR%20owningItem:94925%20OR%20owningItem:94978%20OR%20owningItem:95010%20OR%20owningItem:95012%20OR%20owningItem:95013%20OR%20owningItem:95416%20OR%20owningItem:95446%20OR%20owningItem:95662%20OR%20owningItem:95667%20OR%20owningItem:95668%20OR%20owningItem:95670%20OR%20owningItem:95671%20OR%20owningItem:95673%20OR%20owningItem:95674%20OR%20owningItem:95863%20OR%20owningItem:95864%20OR%20owningItem:95867%20OR%20owningItem:95868%20OR%20owningItem:[95874%20TO%2095879]%20OR%20owningItem:96184%20OR%20owningItem:96436%20OR%20owningItem:96640%20OR%20owningItem:97530%20OR%20owningItem:97807%20OR%20owningItem:98154%20OR%20owningItem:[98565%20TO%2098569]%20OR%20owningItem:[98902%20TO%2098904]%20OR%20owningItem:99431%20OR%20owningItem:99660%20OR%20owningItem:99671%20OR%20owningItem:[99994%20TO%2099996]%20OR%20owningItem:100058%20OR%20owningItem:100202%20OR%20owningItem:100228%20OR%20owningItem:100231%20OR%20owningItem:100252%20OR%20owningItem:100274%20OR%20owningItem:100366%20OR%20owningItem:100840%20OR%20owningItem:102410%20OR%20owningItem:102962%20OR%20owningItem:103219%20OR%20owningItem:103307%20OR%20owningItem:103361%20OR%20owningItem:103379%20OR%20owningItem:[103381%20TO%20103383]%20OR%20owningItem:103385%20OR%20owningItem:103666%20OR%20owningItem:104049%20OR%20owningItem:104051%20OR%20owningItem:104052%20OR%20owningItem:104930%20OR%20owningItem:104933%20OR%20owningItem:105106%20OR%20owningItem:105405%20OR%20owningItem:[105407%20TO%20105411]%20OR%20owningItem:105559%20OR%20owningItem:105563%20OR%20owningItem:105565%20OR%20owningItem:105604%20OR%20owningItem:105605%20OR%20owningItem:105954%20OR%20owningItem:106571%20OR%20owningItem:106665%20OR%20owningItem:106667%20OR%20owningItem:106668%20OR%20owningItem:106670%20OR%20owningItem:106672%20OR%20owningItem:106673%20OR%20owningItem:106676%20OR%20owningItem:106694%20OR%20owningItem:[106722%20TO%20106731]%20OR%20owningItem:108989%20OR%20owningItem:108994%20OR%20owningItem:[109122%20TO%20109124]%20OR%20owningItem:[109380%20TO%20109384]%20OR%20owningItem:109929%20OR%20owningItem:111085%20OR%20owningItem:111086%20OR%20owningItem:111088%20OR%20owningItem:111090%20OR%20owningItem:111536%20OR%20owningItem:111803%20OR%20owningItem:111808%20OR%20owningItem:111811%20OR%20owningItem:111812%20OR%20owningItem:111976%20OR%20owningItem:111977%20OR%20owningItem:[112082%20TO%20112096]%20OR%20owningItem:113440%20OR%20owningItem:114340%20OR%20owningItem:[114649%20TO%20114653]%20OR%20owningItem:[114657%20TO%20114661]%20OR%20owningItem:[114663%20TO%20114669]%20OR%20owningItem:[114671%20TO%20114676]%20OR%20owningItem:114714%20OR%20owningItem:114880%20OR%20owningItem:114881%20OR%20owningItem:114883%20OR%20owningItem:114884%20OR%20owningItem:114991%20OR%20owningItem:114992%20OR%20owningItem:114998%20OR%20owningItem:115119%20OR%20owningItem:115127%20OR%20owningItem:[115268%20TO%20115273]%20OR%20owningItem:115275%20OR%20owningItem:115392%20OR%20owningItem:115393%20OR%20owningItem:116045%20OR%20owningItem:116065%20OR%20owningItem:116238%20OR%20owningItem:116309%20OR%20owningItem:116322%20OR%20owningItem:117375%20OR%20owningItem:117377%20OR%20owningItem:117379%20OR%20owningItem:117382%20OR%20owningItem:[117543%20TO%20117547]%20OR%20owningItem:117652%20OR%20owningItem:118040%20OR%20owningItem:118041%20OR%20owningItem:118131%20OR%20owningItem:118158%20OR%20owningItem:118278%20OR%20owningItem:[118286%20TO%20118288]%20OR%20owningItem:118291%20OR%20owningItem:118295%20OR%20owningItem:[119270%20TO%20119273]%20OR%20owningItem:119834%20OR%20owningItem:[120680%20TO%20120685]%20OR%20owningItem:120933%20OR%20owningItem:120936%20OR%20owningItem:121083%20OR%20owningItem:122318%20OR%20owningItem:122411%20OR%20owningItem:122414%20OR%20owningItem:122416%20OR%20owningItem:122418%20OR%20owningItem:123378%20OR%20owningItem:123379%20OR%20owningItem:126149%20OR%20owningItem:126150%20OR%20owningItem:126490%20OR%20owningItem:[127123%20TO%20127125]%20OR%20owningItem:127160%20OR%20owningItem:127218%20OR%20owningItem:[127853%20TO%20127855]%20OR%20owningItem:128046%20OR%20owningItem:[129449%20TO%20129453]%20OR%20owningItem:129455%20OR%20owningItem:129457%20OR%20owningItem:[129459%20TO%20129463]%20OR%20owningItem:[129465%20TO%20129470]%20OR%20owningItem:129473%20OR%20owningItem:129486%20OR%20owningItem:129487%20OR%20owningItem:129489%20OR%20owningItem:129491%20OR%20owningItem:129507%20OR%20owningItem:129659%20OR%20owningItem:129676%20OR%20owningItem:130655%20OR%20owningItem:130657%20OR%20owningItem:[130659%20TO%20130662]%20OR%20owningItem:130965%20OR%20owningItem:130983%20OR%20owningItem:130987%20OR%20owningItem:130996%20OR%20owningItem:131034%20OR%20owningItem:131047%20OR%20owningItem:131096%20OR%20owningItem:131208%20OR%20owningItem:131915%20OR%20owningItem:132764%20OR%20owningItem:132766%20OR%20owningItem:132767%20OR%20owningItem:132772%20OR%20owningItem:132773%20OR%20owningItem:132775%20OR%20owningItem:132776%20OR%20owningItem:132778%20OR%20owningItem:132782%20OR%20owningItem:132783%20OR%20owningItem:132787%20OR%20owningItem:132790%20OR%20owningItem:132791%20OR%20owningItem:132795%20OR%20owningItem:132797%20OR%20owningItem:132798%20OR%20owningItem:132800%20OR%20owningItem:132803%20OR%20owningItem:132806%20OR%20owningItem:132809%20OR%20owningItem:132814%20OR%20owningItem:132817%20OR%20owningItem:132818%20OR%20owningItem:132820%20OR%20owningItem:[132823%20TO%20132833]%20OR%20owningItem:[132835%20TO%20132840]%20OR%20owningItem:133546%20OR%20owningItem:133547%20OR%20owningItem:[134407%20TO%20134410]))&rows=0&facet=true&facet.sort=count&facet.limit=10&facet.field=owningItem&facet.mincount=1&wt=xslt&tr=journal-landing.xsl

```

daisieh commented:
Another thing we should deal with, but not necessarily right now: the static page `journal-lookup.json` needs to be updated with all ISSNs or be replaced with a dynamic lookup for DryadJournalConcepts.

ghost commented:
@daisieh I'll have to look into the error with Evolution and BMC Evolutionary Biology.

I agree that the journal-lookup.json data should be coming from a DryadJournalConcepts lookup, but I  think it might be better handled in a separate issue/PR.

rnathanday committed:
Fix page load speed; improve download stats loading

- removes Cocoon CInclude approach to parallelizing Solr queries, in
favor of fanning out from the browser
- batches Solr queries to avoid response failures due to overly long
queries
- disables download-stats tabs in browser until data is available

ghost commented:
@daisieh Sorry for the delay in getting this fix done. 
The most recent commit improves page load speed significantly and addresses the issue behind the Evolution and BMC Evol Bio page failures that you pointed out.
The wiki page documentation has been updated.

daisieh commented:
It all seems to work now! I'm merging.
